sdk/resourcemanager/azure-resourcemanager-network/src/test/java/com/azure/resourcemanager/network/NetworkInterfaceOperationsTests.java

```diff
@@ -431,37 +431,110 @@ public void canDeleteNetworkWithServiceCallBack() {
+            .withAddressPrefixes(Arrays.asList("10.0.0.8/29"))
+            .parent()
+            .apply();
+
+        network.refresh();
+        subnet = network.subnets().get(subnetName);
+
+        Assertions.assertEquals(1, subnet.addressPrefixes().size());
+        Assertions.assertEquals("10.0.0.8/29", subnet.addressPrefixes().iterator().next());
+        Assertions.assertNotNull(subnet.addressPrefix());
+
+        // update withAddressPrefix
+        network.update().updateSubnet(subnetName)
+            .withAddressPrefix("10.0.0.0/29")
+            .parent()
+            .apply();
+
+        network.refresh();
+        subnet = network.subnets().get(subnetName);
+
+        Assertions.assertEquals("10.0.0.0/29", subnet.addressPrefix());
+        Assertions.assertEquals(1, subnet.addressPrefixes().size());
+    }
+
     @Test
     public void canListSubnetAvailableIpAddresses() {
         String networkName = generateRandomResourceName("vnet", 10);
         String subnetName = "subnet1";
-        String nicName = generateRandomResourceName("nic", 10);
+        String subnet2Name = "subnet2";
 
         Network network = networkManager.networks()
             .define(networkName)
             .withRegion(Region.US_EAST)
             .withNewResourceGroup(rgName)
-            .withAddressSpace("10.0.0.0/24")
+            .withAddressSpace("10.0.0.0/28")
             .withSubnet(subnetName, "10.0.0.0/29")
             .create();
 
         Subnet subnet = network.subnets().get(subnetName);
         Set<String> availableIps = subnet.listAvailablePrivateIPAddresses();
-        Assertions.assertTrue(availableIps.size() > 0);
+        Assertions.assertFalse(availableIps.isEmpty());
 
-        String availableIp = availableIps.iterator().next();
+        // occupy all available ip addresses
+        for (String ip : availableIps) {
+            networkManager.networkInterfaces()
+                .define(generateRandomResourceName("nic", 10))
+                .withRegion(Region.US_EAST)
+                .withExistingResourceGroup(rgName)
+                .withExistingPrimaryNetwork(network)
+                .withSubnet(subnetName)
+                .withPrimaryPrivateIPAddressStatic(ip)
+                .create();
+        }
 
-        // occupy the available ip address
-        NetworkInterface nic = networkManager.networkInterfaces()
-            .define(nicName)
+        availableIps = subnet.listAvailablePrivateIPAddresses();
+        Assertions.assertTrue(availableIps.isEmpty());
+
+        // define a new subnet with address prefixes
+        network.update().defineSubnet(subnet2Name)
+            .withAddressPrefixes(Arrays.asList("10.0.0.8/29"))
+            .attach()
+            .apply();
+
+        network.refresh();
+
+        // verify it does not affect previous subnet's available ip addresses
+        subnet = network.subnets().get(subnetName);
+        Assertions.assertTrue(subnet.listAvailablePrivateIPAddresses().isEmpty());
```

Comment:
When second subnet is added, result for `listAvailablePrivateIpAddresses` of the first subnet is not affected, meaning `checkIpAddressAvailability` method will correctly identity which subnet the `startIp` belongs to.

Reply:
I see. If backend actually "know" which subnet the query is about, maybe we only need to try 1st item in addressPrefixes?

Reply:
Emmm, let me test that.

Reply:
> I see. If backend actually "know" which subnet the query is about, maybe we only need to try 1st item in addressPrefixes?

Verified.